### PR TITLE
Mscoco transformations to BigML-COCO and documentation changes

### PR DESCRIPTION
**Pull request by mmerce**
Includes the #258 changes and adds checks and tests.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -146,32 +136,26 @@ For a detailed description of authentication instructions on Windows see the
...
-the input field that appears when you click on ``Start`` and hit ``enter``),
-download this `python file <http://python-distribute.org/distribute_setup.py>`_
-and execute it
-
-.. code-block:: bash
-
-    c:\Python27\python.exe distribute_setup.py
-
-After that, you'll be able to install ``pip`` by typing the following command
+To install BigMLer on Windows environments, you'll Python installed.
```
Comment (jaor) [inline README.rst:139]: typo: you'll need

File: README.rst [hunk 2]
```diff
@@ -146,32 +136,26 @@ For a detailed description of authentication instructions on Windows see the
...
+The code has been tested with Python 3.10 and you can create a conda
```
Comment (jaor) [inline README.rst:140]: nit: maybe make "conda" italics if it's usually spelled lowercase? or maybe not, if everybody knows wthat that is..

File: README.rst [hunk 3]
```diff
@@ -146,32 +136,26 @@ For a detailed description of authentication instructions on Windows see the
...
+environment with that Python version or download it from `Python for Windows
+ <http://www.python.org/download/>`_ and install it. In the last case, you'll
```
Comment (jaor) [inline README.rst:142]: In the latter case

File: README.rst [hunk 4]
```diff
@@ -146,32 +136,26 @@ For a detailed description of authentication instructions on Windows see the
...
+ also need too install the ``pip`` tool to install BigMLer.
 
-.. code-block:: bash
-
-    c:\Python27\Scripts\easy_install.exe pip
+To install ``pip``, first you need to open your command line window
```
Comment (jaor) [inline README.rst:145]: a command line terminal window (?)

File: README.rst [hunk 5]
```diff
@@ -180,6 +164,11 @@ issuing
 
 should show BigMLer version information.
 
+Extensions of BigMLer to use images are not supported in Windows by default.
+The libraries needed for those models are not available usually for that
```
Comment (jaor) [inline README.rst:168]: are usually not available

File: bigmler/options/source.py [hunk 6]
```diff
@@ -334,12 +334,16 @@ def get_source_options(defaults=None):
...
             'help': ("Language used to provide the annotations for images."
                      "Annotations are expected to be provided using "
-                     "on file per image. The --train option should point"
+                     "one file per image. The --train option should point"
                      " to the directory that contains both images and"
-                     " the corresponding annotations.")},
+                     " the corresponding annotations, unless some "
+                     " folder attribute is provided in each"
+                     " annotation. In that case it should point to"
+                     " the folder parent directory and --anotations-dir"
```
Comment (jaor) [inline bigmler/options/source.py:345]: folder's

File: bigmler/options/source.py [hunk 7]
```diff
@@ -356,7 +360,15 @@ def get_source_options(defaults=None):
...
             'default': defaults.get('annotations_dir', None),
-            'help': "Directory for individual annotation files."},
+            'help': ("Directory for individual annotation files."
+                     " Used when annotations are provided using "
+                     "one file per image. The --train option should point"
+                     " to the directory that contains both images and"
+                     " the corresponding annotations, unless some "
+                     " folder attribute is provided in each"
+                     " annotation. In that case it should point to"
+                     " the folder parent directory and --anotations-dir"
```
Comment (jaor) [inline bigmler/options/source.py:370]: folder's (IIUC)

File: bigmler/processing/annotations.py [hunk 8]
```diff
@@ -157,16 +161,22 @@ def bigml_metadata(args, images_list=None, new_fields=None):
...
     args.annotations_file = os.path.join(args.output_dir, "annotations.json")
-    filenames = voc_to_cocojson(args.annotations_dir, args,
-                                session_file) \
-        if args.annotations_language == "VOC" else \
-        yolo_to_cocojson(args.annotations_dir, args, session_file)
+    if args.annotations_language == "VOC":
+        filenames = voc_to_cocojson(args.annotations_dir, args, session_file)
+    elif args.annotations_language == "YOLO":
+        filenames = yolo_to_cocojson(args.annotations_dir, args, session_file)
+    elif args.annotations_language == "COCO":
```
Comment (jaor) [inline bigmler/processing/annotations.py:176]: i am neutral about it, but have you considered naming it MSCOCO to be explicit about its not being pure COCO? or is it the case that the only COCO is MSCOCO?

**Review by jaor — APPROVED** [review]
looks good to me, just a few typos and nits

File: bigmler/processing/annotations.py (hunk 8 above)
Comment (mmerce) [reply bigmler/processing/annotations.py:176]: I though so too, but COCO is shorter and from what I saw is most common.

**Pull request merged**